### PR TITLE
Mavlink

### PR DESCRIPTION
**Pull request by otaviobp**
- Fixes some minor issues
- Change socket classes to suport sending packets to different destinations
- Use MAVLink packets to send the RTSP URI so the mavlink-client can select and play video streams

### CODE REVIEW

File: samples/main_sample_mavlink_client.cpp [hunk 1]
```diff
@@ -49,6 +70,70 @@ static bool send_camera_request(Context &ctx, const struct sockaddr_in &addr)
...
 }
 
+static int read_int()
+{
+    char *text = nullptr;
+    int ret, value;
+    ssize_t read;
+    size_t n, len;
+
+begin:
```
Comment (lucasdemarchi) [inline samples/main_sample_mavlink_client.cpp]: `do { } while (1)` ?

**Comment by lucasdemarchi** [comment]
Nicely done! LGTM.

File: src/mavlink_server.cpp [hunk 2]
```diff
@@ -23,46 +23,186 @@
...
+    else
+        _broadcast_addr.sin_addr.s_addr = inet_addr(DEFAULT_MAVLINK_BROADCAST_ADDR);
+    _broadcast_addr.sin_family = AF_INET;
+    _rtsp_server_addr = opt.rtsp_server_addr;
 }
 
 MavlinkServer::~MavlinkServer()
 {
     stop();
+    free(_rtsp_server_addr);
```
Comment (lucasdemarchi) [inline src/mavlink_server.cpp]: nah... it's perfectly fine to call free() with NULL as argument.

File: src/mavlink_server.cpp [hunk 3]
```diff
@@ -23,46 +23,186 @@
 #include "log.h"
 #include "mainloop.h"
 #include "mavlink_server.h"
+#include "util.h"
 
 #define DEFAULT_MAVLINK_PORT 14550
+#define DEFAULT_SYSID 42
```
Comment (otaviobp) [inline src/mavlink_server.cpp]: Not really. Just any number different from 1, 255 and any other system ID being used. There is a way to set it in config file doing:

    [mavlink]
    system_id = 123 #change it to 123

File: src/mavlink_server.cpp (hunk 2 above)
Comment (otaviobp) [inline src/mavlink_server.cpp]: Calling free(NULL) is not a problem, so it is not necessary to check for that.

File: src/mavlink_server.cpp [hunk 4]
```diff
@@ -123,8 +275,10 @@ void MavlinkServer::start()
         return;
     _is_running = true;
 
-    _udp.open(DEFAULT_MAVLINK_BROADCAST_ADDR, DEFAULT_MAVLINK_PORT, false);
-    _udp.set_read_callback([this](const struct buffer &buf) { this->_message_received(buf); });
+    _udp.open(true);
```
Comment (otaviobp) [inline src/mavlink_server.cpp]: it makes sense. I'll add that.

**Comment by otaviobp** [comment]
mnerged

File: src/socket.cpp [hunk 5]
```diff
@@ -19,12 +19,96 @@
...
+
+Socket::Socket()
+    : _read_cb([](const struct buffer &buf, const struct sockaddr_in &sockaddr) {})
+    , _write_buf{0, nullptr}
+{
+}
+
+Socket::~Socket()
+{
+    if (_write_buf.data)
```
Comment (edersondisouza) [inline src/socket.cpp]: Why this check for `NULL` before free?

Comment (otaviobp) [inline src/socket.cpp]: fixed. thanks.

**Pull request closed without merging**